src/vm/threadsuspend.cpp

```diff
@@ -7411,8 +7411,417 @@ void STDCALL OnHijackStructInRegsWorker(HijackArgs * pArgs)
+#ifdef _DEBUG
+        if (g_pConfig->FastGCStressLevel()) {
+            GC_ON_TRANSITIONS(GCOnTransition);
+        }
+#endif // _DEBUG
+
+        // Update the references in the returned struct
+        *((OBJECTREF *)&pArgs->ReturnValue[0]) = oref;
+        *(size_t*)&pArgs->ReturnValue[1] = (size_t)ptr;
+    }
+    GCPROTECT_END();        // trashes ptr here!
+    GCPROTECT_END();        // trashes oref here!
+
+    frame.Pop();
+#else
+    PORTABILITY_ASSERT("OnHijackObjectInteriorPointerWorker not implemented on this platform.");
+#endif //  HIJACK_NONINTERRUPTIBLE_THREADS
+}
+
+inline void SwapReturnRegisters(HijackArgs * pArgs)
+{
+    size_t swap = pArgs->ReturnValue[0];
+    pArgs->ReturnValue[0] = pArgs->ReturnValue[1];
+    pArgs->ReturnValue[1] = swap;
+}
+
+void STDCALL OnHijackRetKindWorker(HijackArgs * pArgs)
+{
+    CONTRACTL{
+        THROWS;
+    GC_TRIGGERS;
+    SO_TOLERANT;
+    } CONTRACTL_END;
+
+    Thread         *thread = GetThread();
+    ReturnKind returnKind = thread->GetHijackReturnKind();
+
+    switch (returnKind)
+    {
+    case RT_Scalar:
+        OnHijackScalarWorker(pArgs);
+        break;
+
+    case RT_Object:
+        OnHijackObjectWorker(pArgs);
+        break;
+
+    case RT_ByRef:
+        OnHijackInteriorPointerWorker(pArgs);
+        break;
+ 
+    //   RT_firstReg_secondReg
+    case RT_Scalar_Obj:
+        SwapReturnRegisters(pArgs);
+        OnHijackObjectWorker(pArgs);
+        SwapReturnRegisters(pArgs);
+        break;
+
+    case RT_Scalar_ByRef:  
+        SwapReturnRegisters(pArgs);
+        OnHijackInteriorPointerWorker(pArgs);
+        SwapReturnRegisters(pArgs);
+        break;
+
+    case RT_Obj_Obj:
+        OnHijackObject2Worker(pArgs);
+        break;
+
+    case RT_ByRef_ByRef:
+        OnHijackInteriorPointer2Worker(pArgs);
+        break;
+
+    case RT_Obj_ByRef:
+        OnHijackObjectInteriorPointerWorker(pArgs);
+        break;
+
+    case RT_ByRef_Obj:
+        SwapReturnRegisters(pArgs);
+        OnHijackObjectInteriorPointerWorker(pArgs);
+        SwapReturnRegisters(pArgs);
+        break;
+
+    default:
+        _ASSERTE(false); // Unexpected case
+        break;
+    }
+}
+
+#endif // FEATURE_MULTIREG_RETURN
+
+VOID * GetHijackAddrFromMethodTable(Thread *pThread, EECodeInfo *codeInfo)
+{
+#ifdef _WIN64
+    // For simplicity, we don't hijack in funclets, but if you ever change that, 
+    // be sure to choose the OnHijack... callback type to match that of the FUNCLET
+    // not the main method (it would probably be Scalar).
+#endif // _WIN64
+
+    ENABLE_FORBID_GC_LOADER_USE_IN_THIS_SCOPE();
+    // Mark that we are performing a stackwalker like operation on the current thread.
+    // This is necessary to allow the signature parsing functions to work without triggering any loads
+    ClrFlsValueSwitch threadStackWalking(TlsIdx_StackWalkerWalkingThread, pThread);
+
+    MethodDesc *methodDesc = codeInfo->GetMethodDesc();
+
+    if (methodDesc == nullptr) {
```

Comment:
I'm not sure how it can be null. 
I wrote this case to mimic the existing behavior in [` Thread::HandledJITCase`](https://github.com/dotnet/coreclr/blob/master/src/vm/threadsuspend.cpp#L7868-L7909) which I refactored out to this function.